boa/src/builtins/typed_array/mod.rs

```diff
@@ -0,0 +1,130 @@
+//! <https://tc39.es/ecma262/#sec-typedarray-objects>
+use crate::{
+    object::{ConstructorBuilder, JsObject, PROTOTYPE},
+    property::Attribute,
+    value::JsValue,
+    BoaProfiler, Context, JsResult,
+};
+use integer_indexed_object::IntegerIndexedObject;
+pub use integer_indexed_object::IntegerIndexedObject;
+
+pub mod int8_array;
+mod integer_indexed_object;
+
+/// The JavaScript `%TypedArray%` object.
+///
+/// <https://tc39.es/ecma262/#sec-%typedarray%-intrinsic-object>
+#[derive(Debug, Clone, Copy)]
+struct TypedArray;
+
+impl TypedArray {
+    const NAME: &'static str = "TypedArray";
+    const LENGTH: usize = 0;
+
+    fn init(context: &mut Context) -> JsObject {
+        let _timer = BoaProfiler::global().start_event(Self::NAME, "init");
+
+        ConstructorBuilder::with_standard_object(
+            context,
+            Self::constructor,
+            context.standard_objects().function_object().clone(),
+        )
+        .name(Self::NAME)
+        .length(Self::LENGTH)
+        .property(
+            "length",
+            0,
+            Attribute::WRITABLE | Attribute::NON_ENUMERABLE | Attribute::PERMANENT,
+        )
+        .build()
+    }
+
+    /// <https://tc39.es/ecma262/#sec-%typedarray%>
+    fn constructor(
+        new_target: &JsValue,
+        args: &[JsValue],
+        context: &mut Context,
+    ) -> JsResult<JsValue> {
+        // 1. Throw a TypeError exception.
+        context.throw_type_error("the TypedArray constructor should never be called directly")
+    }
+}
+
+/// `AllocateTypedArray ( constructorName, newTarget, defaultProto [ , length ] )`
+///
+/// It is used to validate and create an instance of a `TypedArray` constructor. If the `length`
+/// argument is passed, an `ArrayBuffer` of that length is also allocated and associated with the
+/// new `TypedArray` instance. `AllocateTypedArray` provides common semantics that is used by
+/// `TypedArray`.
+///
+/// For more information, check the [spec][spec].
+///
+/// [spec]: https://tc39.es/ecma262/#sec-allocatetypedarray
+fn allocate_typed_array<P>(
+    constructor_name: TypedArrayName,
+    new_target: &JsValue,
+    default_proto: P,
+    length: Option<usize>,
+    context: &mut Context,
+) -> JsResult<JsValue>
+where
+    P: FnOnce() -> JsObject,
+{
+    // 1. Let proto be ? GetPrototypeFromConstructor(newTarget, defaultProto).
+    // TODO: move out "GetPrototypeFromConstructor"
+    let proto = new_target
+        .as_object()
+        .and_then(|obj| {
+            obj.__get__(&PROTOTYPE.into(), obj.clone().into(), context)
+                .map(|o| o.as_object())
+                .transpose()
+        })
+        .transpose()?
+        .unwrap_or_else(default_proto);
```

Comment:
This is a nice API :)
I completely support the idea of extracting it as `GetPrototypeFromConstructor`.

Reply:
Hopefully all those TODOs should be solved before merging, so we need full `ArrayBuffer` support, full Data Block support, full `TypedArray` support and adding the possible methods to `$262` testing object. Feel free to commit something as I will be without a computer for a week or so 😅